openmdao/docs/openmdao_book/features/core_features/options/options.ipynb

```diff
@@ -767,12 +767,111 @@
     "assert_near_equal(expected, c3y)"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Adding a function to pre-process setting an option\n",
+    "\n",
+    "The `OptionsDictionary` has support for custom pre-processing the value before it is set. One potential use for this is to provide a way to convert units while setting an option. The following example shows how to do this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import unittest\n",
+    "\n",
+    "import openmdao.api as om\n",
+    "from openmdao.utils.units import convert_units\n",
+    "\n",
+    "\n",
+    "# TODO: Turn this into a test.\n",
+    "\n",
+    "def units_setter(opt_meta, value):\n",
+    "    \"\"\"\n",
+    "    Check and convert new units tuple into\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    opt_meta : dict\n",
+    "        Dictionary of entries for the option.\n",
+    "    value : any\n",
+    "        New value for the option.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    any\n",
+    "        Post processed value to set into the option.\n",
+    "    \"\"\"\n",
+    "    new_val, new_units = value\n",
+    "    old_val, units = opt_meta['val']\n",
+    "\n",
+    "    converted_val = convert_units(new_val, new_units, units)\n",
+    "    return (converted_val, units)\n",
+    "\n",
+    "\n",
+    "class MyComp(om.ExplicitComponent):\n",
+    "\n",
+    "    def setup(self):\n",
+    "\n",
+    "        self.add_input('x', 3.0)\n",
+    "        self.add_output('y', 3.0)\n",
+    "\n",
+    "    def initialize(self):\n",
+    "        self.options.declare('length', default=(12.0, 'inch'),\n",
+    "                             set_function=units_setter)\n",
+    "\n",
+    "    def compute(self, inputs, outputs):\n",
+    "        length = self.options['length'][0]\n",
+    "\n",
+    "        x = inputs['x']\n",
+    "        outputs['y'] = length * x\n",
+    "\n",
+    "\n",
+    "class MySubgroup(om.Group):\n",
+    "\n",
+    "    def setup(self):\n",
+    "        self.add_subsystem('mass', MyComp())\n",
+    "\n",
+    "\n",
+    "prob = om.Problem()\n",
+    "model = prob.model\n",
+    "\n",
+    "model.add_subsystem('statics', MySubgroup())\n",
+    "\n",
+    "prob.model_options['*'] = {'length': (2.0, 'ft')}\n",
+    "prob.setup()\n",
+    "\n",
+    "prob.run_model()\n",
+    "\n",
+    "print('The following should be 72 if the units convert correctly.')\n",
+    "print(prob.get_val('statics.mass.y'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "hide_input",
+     "hide_output"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "from openmdao.utils.assert_utils import assert_near_equal\n",
+    "assert_near_equal(prob.get_val('statics.mass.y'), 72, 1e-6)\n"
+   ]
```

Comment:
going to need  `allow_assert` tag:
https://github.com/swryan/OpenMDAO/blob/6cf51394dd1f2543c303e7e33dc15c83e081c870/openmdao/docs/openmdao_book/tests/test_jupyter_output_linting.py#L146

Reply:
Looks like I should be using the remove tag instead of the hide tag.

Reply:
better answer